src/datasets.py

```diff
@@ -40,16 +41,23 @@ def __getitem__(self, idx):
 
 
 class CasiaSurfDataset(Dataset):
-    def __init__(self, dir: str = 'data/CASIA_SURF', train: bool = True):
+    def __init__(self, protocol: int, dir: str = '../data/CASIA_SURF', train: bool = True, ):
         txt_metadata = 'train' if train else 'dev_res'
         self.dir = dir
-        self.items = []
-        for i in [1, 2, 3]:
-            file_name = f'4@{i}_{txt_metadata}.txt'
-            with open(os.path.join(dir, file_name), 'r') as file:
-                lines = file.readlines()
-                self.items += [tuple(line.split(' ')) for line in lines]
-        self.transform = Compose([Resize((640, 480)), ToTensor()])
+        file_name = f'4@{protocol}_{txt_metadata}.txt'
+        with open(os.path.join(dir, file_name), 'r') as file:
+            lines = file.readlines()
+            if train:
+                self.items = [tuple(line[:-1].split(' ')) for line in lines]
+            else:
+                self.items = []
+                for line in lines:
+                    dir_name = os.path.join(line[:-1], 'profile')
+                    for file_name in os.listdir(os.path.join(dir, dir_name)):
+                        item = os.path.join(dir_name, file_name)
+                        self.items.append((item, -1))
+
+        self.transform = Compose([Resize((320, 240)), ToTensor()])
```

Comment:
The same comment.
self.transform = transform (transform is coming as parameter to constructor)

Reply:
There is a must-have ToTensor() transform, which converts PIL.Image to torch.Tensor for loss calculation. So if transforms would be acquired from constructor, at any time you have to specify ToTensor() transform to make things work.